cdap-docs/examples-manual/source/examples/data-cleansing.rst

```diff
@@ -0,0 +1,179 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :description: Cask Data Application Platform Data Cleansing Application
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _examples-word-count:
+
+==============
+Data Cleansing
+==============
+
+A Cask Data Application Platform (CDAP) example demonstrating incrementally consuming partitions of a partitioned fileset using MapReduce.
+
+Overview
+========
+
+This application has a MapReduce which processes records from one partitioned file set into another partitioned file set,
+while filtering records that do not match a particular schema.
+
+- The ``DataCleansingService`` writes and serves data from the partitioned file sets.
+- The ``DataCleansingMapReduce`` processes the records, while filtering 'unclean' or invalid records.
+
+Let's look at some of these components, and then run the application and see the results.
+
+The Data Cleansing Application
+------------------------------
+
+As in the other :ref:`examples <examples-index>`, the components
+of the application are tied together by the class ``DataCleansing``:
+
+.. literalinclude:: /../../../cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansing.java
+   :language: java
+   :lines: 27-
+
+Data Storage
+------------
+
+- ``rawRecords`` input PartitionedFileSet of the DataCleansingMapReduce, contains any ingested records.
+- ``cleanRecords`` output PartitionedFileSet, contains only the filtered records.
+- ``consumingState`` stores the state of the DataCleansingMapReduce, such that in each run,
+  it processes only new partitions.
+
+
+DataCleansingService
+--------------------
+
+The service allows writing to the 'rawRecords' and reading from the 'cleanRecords' PartitionedFileSet
+It exposes these endpoints:
+
+- POST ``/rawRecords`` allows for writing to a partition of the 'rawRecords' dataset;
+- GET ``/cleanRecords/{partition key}`` returns the contents of the file of the partition, specified
+  by the given key, of the cleanRecords dataset.
+
+MapReduce over PartitionedFileSet
+---------------------------------
+
+``DataCleansingMapReduce`` is a simple MapReduce that reads from the ``rawRecords`` PartitionedFileSet and writes to
+the ``cleanRecords`` PartitionedFileSet. The ``beforeSubmit()`` method prepares the MapReduce program for this:
+
+- It uses the ``BatchPartitionConsumer`` to specify the partitions to process as input, in order to only process new
+  partitions since its last run.
+- It specifies the output partition to write to based upon the runtime arguments.
```

Comment:
It specifies the output partition that is written to, based upon the supplied runtime arguments.